lib/commands.coffee

```diff
@@ -51,12 +52,26 @@ class Commands
       atom.notifications.addError error.message, {detail: error.stack, dismissable: true}
 
   nextError: ->
-    if not @messages or (next = @messages.next()).done
-      next = (@messages = @linter.views.getMessages().values()).next()
-    return if next.done # There's no errors
-    message = next.value
-    return unless message.filePath
-    return unless message.range
+    if @index?
+      @index++
+    else
+      @index = 0
```

Comment:
Yes and some api like that can take a cursor and return the next and previous error from that structure (excluding the one the cursor it at right now)